tests/functional/conftest.py

```diff
@@ -212,21 +211,32 @@ def _check(
 @pytest.fixture
 def wait_for_sidekiq(gl):
     """
-    Return a helper function to wait until there are no busy sidekiq processes.
+    Return a helper function to wait until there are spare sidekiq processes.
+
+    Because not all Groups can be deleted immediately in GL 16, we will likely have busy Sidekiq jobs
+    set aside for their deletion in 1 days time.
 
     Use this with asserts for slow tasks (group/project/user creation/deletion).
+
+    { ..., 'labels': [], 'concurrency': 20, 'busy': 3}
     """
 
     def _wait(timeout: int = 30, step: float = 0.5, allow_fail: bool = False) -> bool:
         for count in range(timeout):
             time.sleep(step)
-            busy = False
             processes = gl.sidekiq.process_metrics()["processes"]
+            busy_processes = 0
+            total_concurrency = 0
             for process in processes:
-                if process["busy"]:
-                    busy = True
-            if not busy:
+                busy_processes += process["busy"]
+                total_concurrency += process["concurrency"]
+
+            logging.info(f"Busy processes {busy_processes}")
+            #  If we have space concurrency in the process, continue
+            if busy_processes < total_concurrency:
```

Comment:
I'll be honest this looks concerning to me. As I think this will increase our chances of things randomly breaking because the process we wanted to finish is not finished.